[WPE] WPE Platform: add none IM context implementation as fallback

carlosgcampos opened the pull request:
#### 635b3690eaea76a04a790a5ca8b8eb99d7191f0d
<pre>
[WPE] WPE Platform: add none IM context implementation as fallback
<a href="https://bugs.webkit.org/show_bug.cgi?id=275480">https://bugs.webkit.org/show_bug.cgi?id=275480</a>

Reviewed by Nikolas Zimmermann.

We don&apos;t want object new methods that can return nullptr, so better
have a None implementation used as fallback for platforms not implementing IME.
Also make create_view mandatory for the same reason, but in this case
all platform implementations are expected to implement WPEView.

* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayCreateView):
(wpeDisplayCreateInputMethodContext):
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContext.cpp:
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContextNone.cpp: Added.
(wpeInputMethodContextNoneGetPreeditString):
(wpe_input_method_context_none_class_init):
(wpeInputMethodContextNoneNew):
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContextNone.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:

Canonical link: <a href="https://commits.webkit.org/280009@main">https://commits.webkit.org/280009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d846128b0dc4553fe1221bfbb86093a357139a9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55509 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34832 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7976 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58494 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5940 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42455 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6138 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44701 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4076 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57538 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32740 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47839 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25829 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29524 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4084 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51458 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5435 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60085 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30663 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5558 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/52137 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31748 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47909 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51606 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32829 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8179 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31495 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
Great, thanks for the quick fix.

carlosgcampos committed:
[WPE] WPE Platform: add none IM context implementation as fallback
https://bugs.webkit.org/show_bug.cgi?id=275480

Reviewed by Nikolas Zimmermann.

We don't want object new methods that can return nullptr, so better
have a None implementation used as fallback for platforms not implementing IME.
Also make create_view mandatory for the same reason, but in this case
all platform implementations are expected to implement WPEView.

* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayCreateView):
(wpeDisplayCreateInputMethodContext):
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContext.cpp:
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContextNone.cpp: Added.
(wpeInputMethodContextNoneGetPreeditString):
(wpe_input_method_context_none_class_init):
(wpeInputMethodContextNoneNew):
* Source/WebKit/WPEPlatform/wpe/WPEInputMethodContextNone.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:

Canonical link: https://commits.webkit.org/280009@main
